common/src/main/java/tachyon/security/PlainSaslHelper.java

```diff
@@ -67,7 +74,47 @@ public static TTransportFactory getPlainServerTransportFactory(AuthType authType
     return saslFactory;
   }
 
-  // TODO: get client side PLAIN TTransport
-//  public static TTransport getPlainClientTransport(String username, String password,
-//      TTransport wrappedTransport) throws SaslException {}
+  /**
+   * Get a ClientTransport for Plain
+   * @param username User Name of PlainClient
+   * @param password Password of PlainClient
+   * @param wrappedTransport The original Transport
+   * @return Wrapped Transport with Plain
+   * @throws SaslException if an AuthenticationProvider is not found
+   */
+  public static TTransport getPlainClientTransport(String username, String password,
+      TTransport wrappedTransport) throws SaslException {
+    return new TSaslClientTransport("PLAIN", null, null, null, new HashMap<String, String>(),
+        new PlainClientCallbackHandler(username, password), wrappedTransport);
+  }
+
+  /**
+   * A client side callback to put application provided username/password into SASL transport.
+   */
+  public static class PlainClientCallbackHandler implements CallbackHandler {
+
+    private final String mUserName;
+    private final String mPassword;
+
+    public PlainClientCallbackHandler(String userName, String password) {
+      mUserName = userName;
+      mPassword = password;
+    }
+
+    @Override
+    public void handle(Callback[] callbacks) throws IOException, UnsupportedCallbackException {
+      for (Callback callback : callbacks) {
+        if (callback instanceof NameCallback) {
+          NameCallback nameCallback = (NameCallback) callback;
+          nameCallback.setName(mUserName);
+        } else if (callback instanceof PasswordCallback) {
+          PasswordCallback passCallback = (PasswordCallback) callback;
```

Comment:
Hi Qifan, we have disallowed null name or null password at server side, client side may not need to add this check.